graf2d/gpadv7/v7/inc/ROOT/RAttrMap.hxx

```diff
@@ -0,0 +1,179 @@
+/// \file ROOT/RAttrMap.hxx
+/// \ingroup Gpad ROOT7
+/// \author Axel Naumann <axel@cern.ch>
+/// \author Sergey Linev <s.linev@gsi.de>
+/// \date 2017-09-26
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RAttrMap
+#define ROOT7_RAttrMap
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <vector>
+#include <list>
+
+namespace ROOT {
+namespace Experimental {
+
+class RAttrBase;
+
+class RAttrMap {
+
+   friend class RAttrBase;
+
+public:
+
+   enum EValuesKind { kBool, kInt, kDouble, kString };
```

Comment:
Main problem - these many data members will be streamed by I/O and increase size on disk.
It would be grate to implement this functionality with std::variant, but we are far away from support of std::variant in ROOT I/O.